pygmt/io.py

```diff
@@ -39,7 +39,15 @@ def load_dataarray(filename_or_obj, **kwargs):
     if "cache" in kwargs:
         raise TypeError("cache has no effect in this context")
 
-    with xr.open_dataarray(filename_or_obj, **kwargs) as dataarray:
+    if kwargs.get("engine") == "rasterio":
+        import rioxarray
+
+        open_dataarray_func = rioxarray.open_rasterio
+        kwargs.pop("engine")
+    else:
+        open_dataarray_func = xr.open_dataarray
+
+    with open_dataarray_func(filename_or_obj, **kwargs) as dataarray:
         result = dataarray.load()
         _ = result.gmt  # load GMTDataArray accessor information
```

Comment:
There is a segfault when loading the RGB image, see https://github.com/GenericMappingTools/pygmt/actions/runs/7939085789/job/21678737828?pr=2235#step:8:415:

```python-traceback
Fatal Python error: Segmentation fault

Current thread 0x00007f0a18438740 (most recent call first):
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/xarray/backends/file_manager.py", line 234 in close
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/xarray/core/common.py", line 1205 in close
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/xarray/core/common.py", line 1422 in __exit__
  File "/home/runner/work/pygmt/pygmt/pygmt/io.py", line 50 in load_dataarray
  File "/home/runner/work/pygmt/pygmt/pygmt/datasets/load_remote_dataset.py", line 439 in _load_remote_dataset
  File "/home/runner/work/pygmt/pygmt/pygmt/helpers/decorators.py", line 776 in new_module
  File "/home/runner/work/pygmt/pygmt/pygmt/datasets/earth_daynight.py", line 57 in load_blue_marble
  File "/home/runner/work/pygmt/pygmt/pygmt/tests/test_grdimage_image.py", line 17 in fixture_xr_image
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/fixtures.py", line 902 in call_fixture_func
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/fixtures.py", line 1123 in pytest_fixture_setup
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/pluggy/_callers.py", line 77 in _multicall
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/pluggy/_manager.py", line 115 in _hookexec
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/pluggy/_hooks.py", line 493 in __call__
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/fixtures.py", line 1069 in execute
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/fixtures.py", line 693 in _compute_fixture_value
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/fixtures.py", line 607 in _get_active_fixturedef
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/fixtures.py", line 585 in getfixturevalue
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/fixtures.py", line 566 in _fillfixtures
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/python.py", line 1795 in setup
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/runner.py", line 494 in setup
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/runner.py", line 157 in pytest_runtest_setup
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/pluggy/_callers.py", line 77 in _multicall
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/pluggy/_manager.py", line 115 in _hookexec
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/pluggy/_hooks.py", line 493 in __call__
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/runner.py", line 262 in <lambda>
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/runner.py", line 341 in from_call
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/runner.py", line 261 in call_runtest_hook
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/runner.py", line 222 in call_and_report
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/runner.py", line 127 in runtestprotocol
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/runner.py", line 114 in pytest_runtest_protocol
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/pluggy/_callers.py", line 77 in _multicall
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/pluggy/_manager.py", line 115 in _hookexec
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/pluggy/_hooks.py", line 493 in __call__
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/main.py", line 350 in pytest_runtestloop
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/pluggy/_callers.py", line 77 in _multicall
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/pluggy/_manager.py", line 115 in _hookexec
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/pluggy/_hooks.py", line 493 in __call__
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/main.py", line 325 in _main
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/main.py", line 271 in wrap_session
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/main.py", line 318 in pytest_cmdline_main
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/pluggy/_callers.py", line 77 in _multicall
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/pluggy/_manager.py", line 115 in _hookexec
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/pluggy/_hooks.py", line 493 in __call__
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/config/__init__.py", line 169 in main
  File "/home/runner/micromamba/envs/pygmt/lib/python3.12/site-packages/_pytest/config/__init__.py", line 192 in console_main
  File "/home/runner/micromamba/envs/pygmt/bin/pytest", line 10 in <module>

Extension modules: markupsafe._speedups, yaml._yaml, numpy.core._multiarray_umath, numpy.core._multiarray_tests, numpy.linalg._umath_linalg, numpy.fft._pocketfft_internal, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, matplotlib._c_internal_utils, PIL._imaging, matplotlib._path, kiwisolver._cext, pyarrow.lib, pyarrow._hdfsio, pandas._libs.tslibs.np_datetime, pandas._libs.tslibs.dtypes, pandas._libs.tslibs.base, pandas._libs.tslibs.nattype, pandas._libs.tslibs.timezones, pandas._libs.tslibs.ccalendar, pandas._libs.tslibs.fields, pandas._libs.tslibs.timedeltas, pandas._libs.tslibs.tzconversion, pandas._libs.tslibs.timestamps, pandas._libs.properties, pandas._libs.tslibs.offsets, pandas._libs.tslibs.strptime, pandas._libs.tslibs.parsing, pandas._libs.tslibs.conversion, pandas._libs.tslibs.period, pandas._libs.tslibs.vectorized, pandas._libs.ops_dispatch, pandas._libs.missing, pandas._libs.hashtable, pandas._libs.algos, pandas._libs.interval, pandas._libs.lib, pandas._libs.ops, pyarrow._compute, pandas._libs.arrays, pandas._libs.tslib, pandas._libs.sparse, pandas._libs.indexing, pandas._libs.index, pandas._libs.internals, pandas._libs.join, pandas._libs.writers, pandas._libs.window.aggregations, pandas._libs.window.indexers, pandas._libs.reshape, pandas._libs.groupby, pandas._libs.json, pandas._libs.parsers, pandas._libs.testing, cftime._cftime, _brotli, multidict._multidict, yarl._quoting_c, aiohttp._helpers, aiohttp._http_writer, aiohttp._http_parser, aiohttp._websocket, frozenlist._frozenlist, matplotlib._image, rasterio._version, rasterio._err, rasterio._filepath, rasterio._env, rasterio._transform, rasterio._base, rasterio.crs, rasterio._features, rasterio._warp, rasterio._io, psutil._psutil_linux, psutil._psutil_posix, pyproj._compat, pyproj._datadir, pyproj._network, pyproj._geod, pyproj.list, pyproj._crs, pyproj.database, pyproj._transformer, pyproj._sync, shapely.lib, shapely._geos, shapely._geometry_helpers, netCDF4._netCDF4, PIL._imagingmath, fiona._err, fiona._env, fiona.crs, fiona._geometry, fiona.schema, fiona.ogrext (total: 103)
Segmentation fault (core dumped)
make: *** [Makefile:38: _runtest] Error 139
../pygmt/tests/test_grdimage_image.py::test_grdimage_image_dataarray 
```

If I comment out the `_ = result.gmt` line locally, the `test_grdimage_image_dataarray` check will pass, but there will be a segfault later at `test_grdimage_image_dataarray_unsupported_dtype`. Not sure why.

Reply:
> If I comment out the `_ = result.gmt` line locally, the `test_grdimage_image_dataarray` check will pass, but there will be a segfault later at `test_grdimage_image_dataarray_unsupported_dtype`. Not sure why.

Hmm, so I removed the GMTAccessor loading in fc060eaca825d2dba5618b2cf29bfe63e6e4d529 when the Blue Marble GeoTIFF is loaded using `rioxarray.open_rasterio`, and it seems like the segfault disappears on CI, though I still get a segfault if I run just `test_grdimage_image_dataarray` locally...

Reply:
Note to self: using `GMT_OUT|GMT_IS_REFERENCE` for images seems to help to avoid the segfault. See commit 6eabbf4fbab240843c0c804a2a6d16d09313c636, https://github.com/GenericMappingTools/pygmt/pull/3338/commits/c01d6abe524febca3d2454bf74f0f3a866fa87ee (#3338) and https://github.com/GenericMappingTools/pygmt/pull/3115/commits/003383da831be4c82c89dc4187266d0abf0555ec (#3115) where this has been applied.